index.d.ts

```diff
@@ -11,7 +11,7 @@ export type Input = Request | URL | string;
 export type BeforeRequestHook = (
 	input: Input,
 	options: NormalizedOptions,
-) => void | Promise<void>;
+) => Response | void | Promise<void>;
```

Comment:
We also need `Promise<Response>`, if I'm not mistaken. Would be good to add a test case for it in the type tests, too.

Reply:
Good catch, updating.